RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronIDProducer.cc

```diff
@@ -187,22 +82,39 @@ void LowPtGsfElectronIDProducer::produce(edm::StreamID, edm::Event& event, const
   }
 
   // Retrieve GsfElectrons from Event
-  edm::Handle<reco::GsfElectronCollection> gsfElectrons;
+  edm::Handle<edm::View<reco::GsfElectron> > gsfElectrons;
   event.getByToken(gsfElectrons_, gsfElectrons);
   if (!gsfElectrons.isValid()) {
     edm::LogError("Problem with gsfElectrons handle");
   }
 
+  // ElectronSeed unbiased BDT
+  edm::Handle<edm::ValueMap<float> > unbiasedH;
+  event.getByToken(unbiased_, unbiasedH);
+  if (!unbiasedH.isValid()) {
+    edm::LogError("Problem with unbiased handle");
```

Comment:
is it possible that you are using a potentially invalid handle now on line [113](https://github.com/cms-sw/cmssw/pull/31080/files#diff-bd851c5e74fefd74693080a4e490e778R113) below? Would it make sense to abort the workflow if the correct collection was not found?

Reply:
As well as LogError, I now also `throw cms::Exception`. Is this "best practice" in this scenario?
I access two further handles earlier in the produce() method, so I have done the same here.